File: demo-vk-rest-scraping.java/src/main/java/demo/json/CountriesFile.java
package demo.json;

import demo.file.TextFile;
import demo.model.CountriesList;

public class CountriesFile {

    private final TextFile textFile;

    public CountriesFile(String fileName) {
        this.textFile = new TextFile(fileName);
    }

    public CountriesList load() {
        return CountriesSerializer.fromJson(textFile.load());
    }

    public void save(CountriesList countries) {
        textFile.save(CountriesSerializer.toJson(countries));
    }
}

File: demo-vk-rest-scraping.kotlin/src/main/kotlin/demo/json/CountriesFile.kt
package demo.json

import demo.model.CountriesList
import demo.file.TextFile

class CountriesFile(fileName: String) {

    private val textFile: TextFile

    init {
        this.textFile = TextFile(fileName)
    }

    fun load(): CountriesList {
        return CountriesSerializer.fromJson(textFile.load())
    }

    fun save(countries: CountriesList) {
        textFile.save(CountriesSerializer.toJson(countries))
    }
}
